dockerfiles/Dockerfile

```diff
@@ -141,6 +144,9 @@ RUN echo "deb http://deb.debian.org/debian bookworm-backports main" >> /etc/apt/
     liblzma5 \
     libyaml-0-2 \
     libcap2 \
+    libboost-system-dev \
+    libboost-thread-dev \
+    libboost-filesystem-dev \
```

Comment:
This is including the dev libraries in the runtime image - we should just have the minimal set of libraries needed for running fluent bit, not those for building it.

Reply:
Hi @patrick-stephens, I tried to take these out, leaving only the entries above (line 57). But when I did, I was not able to build the SOME/IP input plugin and its dependent libraries from the dev container.

Likewise when I tried to replace these in this section with the "non-dev" version (`libboost-filesystem1.74.0`) the header files are not installed in the container and the build fails as well.

Please advise if there is something I am doing wrong here.

Reply:
Yeah I think you need to look at why, the dev libraries should only be in the build container which is fine. At runtime we need the minimal set of dependencies required to run with, otherwise we end up opening a big bag of security worms with extra dependencies.

Header files are not needed at runtime, they should not be in the runtime image.

So I think you need to ensure the `builder` stage has the dev libraries to compile everything but then the `dep-extractor` stage only includes the runtime libraries.